sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -201,9 +201,10 @@ export function createTokenCycler(
     //   step 1.
     //
 
-    // IF the tenantId passed in token options is different to the one we have, we need to
-    // refresh the token with the new tenantId.
-    const mustRefresh = tenantId !== tokenOptions.tenantId || cycler.mustRefresh;
+    // If the tenantId passed in token options is different to the one we have
+    // Or if we are in claim challenge and the token was rejected and a new access token need to be issued, we need to
+    // refresh the token with the new tenantId or token.
+    const mustRefresh = tenantId !== tokenOptions.tenantId || !!tokenOptions.claims || cycler.mustRefresh;
```

Comment:
@jeremymeng, I'm wondering if we should have an option that can set `mustRefresh` to true instead of adding more logic around specific contents of the token in the cycler. This way the individual callbacks can make the decision about forcing refresh or not. What do you think?

Reply:
It is usually clearer if you do `Boolean(tokenOptions.claims)`

Reply:
@MaryGao can you please add some tests to validate this behavior? I would like to see specifically a test that validates that the token doesn't get refreshed when there is no need to refresh but we still have claims